code/iaas/logic/src/main/resources/META-INF/cattle/core-process/spring-core-processes-context.xml

```diff
@@ -144,4 +144,25 @@
     <!-- Cluster host map -->
     <process:process name="clusterhostmap.create" resourceType="clusterHostMap" start="requested" transitioning="creating" done="created" />
     <process:process name="clusterhostmap.remove" resourceType="clusterHostMap" start="created,creating" transitioning="removing" done="removed" />
+
+    <!-- Service Discovery Environment -->
+    <process:process name="environment.create" resourceType="environment" start="requested" transitioning="activating" done="active" />
+    <process:process name="environment.activateservices" resourceType="environment" start="active" transitioning="updating-active" done="active" />
+    <process:process name="environment.deactivateservices" resourceType="environment" start="active" transitioning="updating-active" done="active" />
```

Comment:
@sonchang I'm changing the logic - activate/deactivate services will become actionHandlers and can be executed on the environment being in any state.